enable/examples/demo/enable/sierpinski_zoom.py

```diff
@@ -0,0 +1,203 @@
+# (C) Copyright 2005-2021 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+"""
+Example application for exploring Sierpinski's Triangle.
+"""
+import numpy as np
+
+from traits.api import (
+    HasTraits,
+    Instance,
+    Int,
+    observe,
+    Range,
+    Property
+)
+from traitsui.api import Item, UItem, View
+from enable.api import (
+    AbstractOverlay, Canvas, Component, ComponentEditor, Viewport, Scrolled
+)
+from enable.tools.api import ViewportPanTool
+
+SQRT3 = np.sqrt(3)
+
+
+class SierpinskiTriangle(AbstractOverlay):
+
+    base_width = Int()
+
+    iterations = Int()
+
+    def overlay(self, component, gc, view_bounds=None, mode="default"):
+        # draw the base triangle
+        gc.translate_ctm(0., 0.)
+        gc.set_fill_color((1.0, 1.0, 1.0, 1.0))
+        gc.move_to(0, 0)
+        gc.line_to(self.base_width, 0)
+        gc.line_to(self.base_width/2, SQRT3*(self.base_width/2))
+        gc.line_to(0, 0)
+        gc.fill_path()
+
+        gc.set_fill_color((0.0, 0.0, 0.0, 1.0))
+
+        gc.begin_path()
+        path = gc.get_empty_path()
+
+        if self.iterations >= 1:
+            with gc:
+                self.add_triangle_to_path(
+                    path,
+                    (self.base_width/4, (self.base_width/4)*SQRT3),
+                    self.base_width/2
+                )
```

Comment:
Why use `with gc` at this level? You don't appear to be modifying the state.

Reply:
Ah you are right, this just didn't get thrown out when I updated the code.  Previously it had modified state.